zap/src/main/java/org/parosproxy/paros/core/scanner/AppParametersBuilder.java

```diff
@@ -0,0 +1,37 @@
+package org.parosproxy.paros.core.scanner;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * {@code AppParametersBuilder} provides an easy way to build {@code AppParameters} and validates
+ * the provided inputs.
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class AppParametersBuilder {
+
+    private List<AppParameter> appParameters = new ArrayList<>();
+    private Set<Integer> nameValuePairPositions = new HashSet<>();
+    private final Logger logger = LogManager.getLogger(this.getClass());
+
+    public AppParametersBuilder addAppParameter(
+            NameValuePair nameValuePair, String param, String value, AppParamValueType operation) {
+        if (!nameValuePairPositions.contains(nameValuePair.getPosition())) {
```

Comment:
@thc202 I have made the condition on the position, considering it is unique and also param can be null too.

Reply:
Sounds good.

Reply:
But it should allow to replace, that way the scan rules can reuse the builder to craft another message with slightly changes.

Reply:
hmm sure.